platforms/documentation/docs/src/docs/userguide/dep-man/basics/dependency_configurations.adoc

```diff
@@ -0,0 +1,145 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[dependency-configurations]]
+= 2. Dependency Configurations
+
+Every dependency declared for a Gradle project applies to a specific scope.
+
+For example, some dependencies should be used for compiling source code whereas others only need to be available at runtime:
+
+[source, kotlin]
+.build.gradle.kts
+----
+dependencies {
+    implementation("com.google.guava:guava:30.0-jre") // Needed to compile and run the app
+    runtimeOnly("org.slf4j:slf4j-simple:2.0.13") // Only needed at runtime
+}
+----
+
+Dependency configurations are a way to define different sets of dependencies for different purposes within a project.
+They determine how and when dependencies are used in various stages of the build process.
+
+Configurations are a fundamental part of dependency resolution in Gradle.
+
+[[sec:what-are-dependency-configurations]]
+== Understanding dependency configurations
+
+Gradle represents the scope of a dependency with the help of a link:{groovyDslPath}/org.gradle.api.artifacts.Configuration.html[Configuration].
+Every configuration can be identified by a unique name.
+
+Many Gradle plugins add pre-defined configurations to your project.
+
+The Java Library plugin is used to define a project that produces a Java library.
+The plugin adds many configurations.
+These configurations represent the various classpaths needed for source code compilation, executing tests, and more:
+
+[cols="~,~,~"]
+|===
+|Configuration Name |Description |Used to:
+
+|`implementation`
+|Dependencies required for both compilation and runtime.
+|Declare Dependencies
+
+|`compileOnly`
+|Dependencies needed only for compilation, not included in runtime or publication.
+|Declare Dependencies
+
+|`runtimeOnly`
+|Dependencies needed only at runtime, not included in the compile classpath.
+|Declare Dependencies
+
+|`compileClasspath`
+|Combines `implementation` and `compileOnly` dependencies, used for compiling source code.
+|Resolve Dependencies
+
+|`runtimeClasspath`
+|Combines `implementation` and `runtimeOnly` dependencies, used for running the application.
+|Resolve Dependencies
+
+|`testImplementation`
+|Dependencies required for compiling and running tests.
+|Declare Dependencies
+
+|`testCompileOnly`
+|Dependencies needed only for test compilation.
+|Declare Dependencies
+
+|`testRuntimeOnly`
+|Dependencies needed only for running tests.
+|Declare Dependencies
+
+|`testCompileClasspath`
+|Combines `testImplementation` and `testCompileOnly` dependencies, used for compiling test sources.
+|Resolve Dependencies
+
+|`testRuntimeClasspath`
+|Combines `testImplementation` and `testRuntimeOnly` dependencies, used for running tests.
+|Resolve Dependencies
+
+|`apiElements`
+|Combines .
+|Consume Dependencies
+
+|`runtimeElements`
+|Combines .
+|Consume Dependencies
```

Comment:
This whole chapter is called "Dependency configurations" though these are resolvable and consumable configurations. I don't think we should mention these (in addition to `runtimeClasspath` and `compileClasspath`).

Reply:
Removed the configs.